python/paddle/fluid/tests/unittests/test_batch_norm_mkldnn_op.py

```diff
@@ -0,0 +1,164 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import numpy as np
+import paddle.fluid.core as core
+from paddle.fluid.op import Operator
+import paddle.fluid as fluid
+from op_test import OpTest
+from paddle.fluid.framework import grad_var_name
+from test_batch_norm_op import TestBatchNormOpInference, TestBatchNormOpTraining, _reference_training, _reference_grad
+
+
+class TestMKLDNNBatchNormOpTraining(TestBatchNormOpTraining):
```

Comment:
Could the batch_norm test like this:
https://github.com/PaddlePaddle/Paddle/blob/5e13314dc1a505ec4a7b2f36e826e1aaf4b2b540/python/paddle/fluid/tests/unittests/test_conv2d_mkldnn_op.py#L20-L22

Only set `init_kernel_type`, and would run mkldnn_batch_norm kernel.

Reply:
I corrected this part. I introduced `init_kernel_type` method in batch norm test cases, that set `use_mkldnn` variable.

Reply:
If you have `init_kernel_type` method, could line 29-148 be removed?